kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemView.java

```diff
@@ -118,7 +118,7 @@ private FileSystemView(FileSystemView<?> view, Schema schema, Class<E> type) {
       writer = PartitionedDatasetWriter.newWriter(this);
     } else {
       writer = FileSystemWriter.newWriter(
-          fs, root, -1, -1 /* get from descriptor */, dataset.getDescriptor());
+          fs, root, -1, -1 /* get from descriptor */, dataset.getDescriptor(), this.getSchema());
```

Comment:
The `AbstractRefinableView#getSchema` method returns the accessor's read schema. That's useful for callers, but the writers should be initialized using `getAccessor().getWriteSchema()`. The two schemas can differ slightly for reflected records.